src/sinks/util/encoding/macros.rs

```diff
@@ -0,0 +1,236 @@
+/// Generates a customized encoding configuration enum for sinks.
+///
+/// ## Purpose
+///
+/// While there are existing encoding enums, such as `StandardEncodings`, some sinks only support a certain subset
+/// of those actual encodings, in terms of what the downstream service/API will accept. This macro allows a caller to
+/// craft a customized encoding enum that only supports the actual encodings they specify.
```

Comment:
Have we considered dropped `StandardEncoding` and forcing the caller to always specify what is supported? As we grow the number of supported encoders, should the "standard" ones also grow. I appreciate the explicit nature of defining them at the sink, but that's also a greater number of places that need updating as we add more widely usable encoders.

Reply:
This will be handled by Pablo's work in #13518 where there will no longer be this encoding enum sort of thing, and all components will get access to all available decoders or encoders.

Once his work lands, we can/should definitely drop `StandardEncodings`, and all other similar encoding enums, with much haste.